src/main/java/com/cisco/trex/stateless/TRexClient.java

```diff
@@ -380,6 +381,68 @@ public void removeRxFilters(int portIndex, int profileId) {
     callMethod("remove_rx_filters", payload);
   }
 
+  /**
+   * Wait until traffic on specified port(s) has ended
+   *
+   * @param timeoutInSecounds
+   * @param rxDelayMs Time to wait (in milliseconds) after last packet was sent, until RX filters
+   *     used for measuring flow statistics and latency are removed. This value should reflect the
+   *     time it takes packets which were transmitted to arrive to the destination. After this time,
+   *     RX filters will be removed, and packets arriving for per flow statistics feature and
+   *     latency flows will be counted as errors.
+   * @param ports Ports on which to execute the command
+   */
+  public void waitOnTrafficToFinish(int timeoutInSecounds, int rxDelayMs, Port... ports) {
+    Map<Port, Boolean> portTrafficMap = new HashMap<>();
+    long endTime = System.currentTimeMillis() + timeoutInSecounds * 1000;
+
+    for (Port port : ports) {
+      portTrafficMap.put(port, true);
+    }
+    while (portTrafficMap.containsValue(true)) {
+      for (Entry<Port, Boolean> entry : portTrafficMap.entrySet()) {
+        if (entry.getValue()) { // port is still true, meaning still running traffic
+          // set port to false if traffic is stopped(IDLE)
+          entry.setValue(!getPortStatus(entry.getKey().getIndex()).get().getState().equals("IDLE"));
+        }
+      }
+      if (System.currentTimeMillis() > endTime) {
+        break;
```

Comment:
1. How to handle the timeout happening? should it throw an exception or logging it.
2. Does not the "timeout" contain the "rxDelayMs" time? it only mean the timeout setting for sending packets on ports.

Reply:
i think we could add a logging of it atleast.
rx delay ms is the time after port have stopped sending traffic and we wait to remove the rx filter, so it means the time it takes for last packet to arrive at destination port.
Another thing that is missing here is a short wait between port status checks.

Reply:
fair enough, please go with update then merging code.